### PR TITLE
Leverage more of angular material for a style refresh

### PR DESCRIPTION
**Pull request by milesmatthias**
Made everything a little more stylized with the material module that's been included.

A couple known issues/improvements I can still address:
- [ ] `mat-toolbar` children in the nav aren't initialized with mat-toolbar styles. You have to click on one of the nav bars to get the styles applied for some reason
- [ ] use [`mat-color`](https://material.angular.io/guide/theming-your-components) from the deep-purple built in theme instead of the raw hex values
- [x] put svg files for bazel & github into the repo instead of using hosted urls
- [x] add menu for small devices

screen shot 2018-12-04 at 5 39 54 pm
screen shot 2018-12-04 at 5 40 08 pm

### CODE REVIEW

**Comment by milesmatthias** [comment]
This is for issue #258

**Comment by gregmagolan** [comment]
Thanks @milesmatthias. That looks great! Rebasing to latest master should fix the CI failure that was caused by event-stream@3.3.6 being removed from npm.

**Comment by gregmagolan** [comment]
Also, running `yarn buildifier` should fix the lint failure.

**Comment by chrislovecnm** [comment]
Should we have "Bazel Example" or "Angulal Bazel Example"  as the name in the header?

File: src/app.component.html [hunk 1]
```diff
@@ -1,4 +1,19 @@
...
+    <!-- TODO replace src with local svg -->
+    <a href="https://bazel.build" class="mat-button">
+      <img class="nav-logo" src="https://bazel.build/images/bazel-navbar.svg">
+    </a>
+    <a routerLink="/" class="mat-button">Home</a>
+    <a routerLink="/todos" class="mat-button">Todos</a>
+    <span class="flex-spacer"></span>
+    <a href="https://github.com/alexeagle/angular-bazel-example" class="mat-button">
+      <!-- TODO replace src with local svg -->
+      <img class="nav-logo" src="https://material.angular.io/assets/img/homepage/github-circle-white-transparent.svg">
```
[the remark this answers is not in the document]
Comment (milesmatthias) [reply src/app.component.html]: Nope, it's a TODO to move the SVG to a local file. Trick there is setting up the bazel function to serve the static image.

File: src/index.html [hunk 2]
```diff
@@ -6,6 +6,7 @@
     <base href="/">
     <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
     <link href="/deeppurple-amber.css" rel="stylesheet">
```
[the remark this answers is not in the document]
Comment (milesmatthias) [reply src/index.html]: Nope, that's a TODO as well.

File: src/BUILD.bazel [hunk 3]
```diff
@@ -120,5 +121,6 @@ history_server(
         ":copy_material_theme",
         ":copy_systemjs",
         ":copy_zonejs",
+        "//src/images:image_assets",
```
Comment (chrislovecnm) [inline src/BUILD.bazel]: Do we need the full path on this “//src”??

Comment (milesmatthias) [reply src/BUILD.bazel]: Yup, we need the full path here. Did you have something else in mind?

**Comment by chrislovecnm** [comment]
@alexeagle PTAL

File: src/app.component.html [hunk 4]
```diff
@@ -1,4 +1,19 @@
-<mat-toolbar>
-  <nav><a routerLink="/">Home</a> | <a routerLink="/todos">Todos</a></nav>
-</mat-toolbar>
-<router-outlet></router-outlet>
+<nav>
+  <mat-toolbar>
+    <!-- TODO replace src with local svg -->
```
[the remark this answers is not in the document]
Comment (alexeagle) [reply src/app.component.html]: sure, /cc @mgechev

**Pull request merged**